src/chunk_processors.py

```diff
@@ -181,3 +196,151 @@ def __call__(self, chunks: Sequence[Chunk]) -> Sequence[Chunk]:
             chunk.heading = current_subheading or current_heading
 
         return new_chunks
+
+
+class RemoveRegexPattern(PipelineComponent):
+    """
+    Remove text from chunks that matches a regex pattern.
+
+    If the chunk's entire text matches the regex pattern, remove the chunk. If it
+    just contains text with the regex pattern, replace the pattern with the text
+    specified.
+    """
+
+    def __init__(self, pattern: str, replace_with: str) -> None:
+        self.pattern = pattern
+        self.replace_with = replace_with
+
+    def __call__(self, chunks: list[Chunk]) -> list[Chunk]:
+        """Run regex pattern removal."""
+        if not hasattr(self, "pattern"):
+            raise ValueError("No pattern was set. Please set a pattern in __init__.")
+
```

Comment:
I can't imagine a way of getting here with a missing `pattern` 🤔 
have you come across one of these in the wild?

Reply:
no, this was a leftover from a previous distracted implementation 🙄  thank you!